libbeat/processors/dns/cache_test.go

```diff
@@ -98,4 +100,19 @@ func TestCache(t *testing.T) {
 		assert.EqualValues(t, 3, c.stats.Hit.Get())
 		assert.EqualValues(t, 3, c.stats.Miss.Get()) // Cache miss.
 	}
+
+	// Cache returned TTL=0 with MinTTL.
+	ptr, err = c.LookupPTR(gatewayIP + "2")
+	if assert.NoError(t, err) {
+		assert.EqualValues(t, gatewayName, ptr.Host)
+		// TTL counts down while in cache.
```

Comment:
I think the TTL returned in the LookupPTR response should reflect the time to live in the cache. So since this was a cache miss it should have the MinTTL value. This makes the response consistent with what is returned when there is a cache hit. As it stands now the next LookupPTR call would return a cache's TTL since it's computed.

https://github.com/elastic/beats/blob/95626b8f1690344312c0831ab2bdcbccffe4d089/libbeat/processors/dns/cache.go#L72